learn-arai/app/hooks/useAuth.ts

```diff
@@ -0,0 +1,75 @@
+'use client';
+
+// this react hook responsible for checking if user is logged in.
+import { useEffect, useState } from 'react';
+
+import { useLocalStorage } from './useLocalStorage';
+import { useUser } from './useUser';
+
+export const useAuth = () => {
+    const { addUser, removeUser, user, getUserFromLocalStorage, setUser } =
+        useUser();
+    const { getItem } = useLocalStorage();
+    const [isAuthenticated, setIsAuthenticated] = useState<boolean>();
+
+    useEffect(() => {
+        checkSession();
+    });
+
+    const signIn = async (credentials: FormData) => {
+        const response = await fetch('http://localhost:3000/auth/sign-in', {
+            method: 'POST',
+            body: credentials,
+            credentials: 'include',
+        });
+
+        const data = await response.json();
+        const message = data.message;
+        const status = data.status;
+
+        if (status == 'success') {
+            const email = credentials.get('email')!.toString();
+            addUser({ email: email });
+        }
```

Comment:
if login isn't successful maybe `removeUser()` just to make sure

Reply:
understood